sdk/ai/Azure.AI.Projects/src/Custom/MemoryStores/AIProjectMemoryStoresOperations.cs

```diff
@@ -141,4 +141,44 @@ public virtual ClientResult<MemoryUpdateResult> UpdateMemories(string memoryStor
         ClientResult protocolResult = UpdateMemories(memoryStoreName, BinaryContent.Create(ModelReaderWriter.Write(options, ModelSerializationExtensions.WireOptions, AzureAIProjectsContext.Default)), cancellationToken.ToRequestOptions());
         return ClientResult.FromValue((MemoryUpdateResult)protocolResult, protocolResult.GetRawResponse());
     }
+
+    /// <summary> Create a memory store. </summary>
+    /// <param name="name"> The name of the memory store. </param>
+    /// <param name="definition"> The memory store definition. </param>
+    /// <param name="description"> A human-readable description of the memory store. </param>
+    /// <param name="metadata"> Arbitrary key-value metadata to associate with the memory store. </param>
+    /// <param name="cancellationToken"> The cancellation token that can be used to cancel the operation. </param>
+    /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="definition"/> is null. </exception>
+    /// <exception cref="ArgumentException"> <paramref name="name"/> is an empty string, and was expected to be non-empty. </exception>
+    /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
+    public virtual ClientResult<MemoryStore> CreateMemoryStore(string name, MemoryStoreDefinition definition, string description = default, IDictionary<string, string> metadata = default, CancellationToken cancellationToken = default)
+    {
+        Argument.AssertNotNullOrEmpty(name, nameof(name));
+        Argument.AssertNotNull(definition, nameof(definition));
+
+        metadata ??= new Dictionary<string, string>();
```

Comment:
Oh, boy. I remember this problem. Optional.IsDefined() doesn't do what it should, right?

Let's address this by customizing the constructor for `InternalCreateMemoryStoreRequest(string, string, IDictionary<string, string>, string, IDictionary<string, BinaryData>)` to initialize `Metadata` there -- to `new ChangeTrackingDictionary<string, string>`. Then we won't need to customize the method.

IIRC, not using `ChangeTrackingDictionary` will cause the property to always serialize, even with an empty collection -- which isn't an error in this case but it's also not optimal.

Reply:
Yes, it does not check value for `null` Undefined means that it is `ChangeTrackingDictionary` AND it has flag `IsUndefined`
```C#
public static bool IsCollectionDefined<TKey, TValue>(IDictionary<TKey, TValue> collection)
{
    return !(collection is ChangeTrackingDictionary<TKey, TValue> changeTrackingDictionary && changeTrackingDictionary.IsUndefined);
}
```

Fixed.